internal/dag/gatewayapi_processor.go

```diff
@@ -1976,6 +1981,81 @@ func (p *GatewayAPIProcessor) httpClusters(routeNamespace string, backendRefs []
 			continue
 		}
 
+		var upstreamValidation *PeerValidationContext
+		var backendRefGroup gatewayapi_v1alpha2.Group
+		if backendRef.Group != nil {
+			backendRefGroup = *backendRef.Group
+		}
+
+		var backendRefKind gatewayapi_v1alpha2.Kind
+		if backendRef.Kind != nil {
+			backendRefKind = *backendRef.Kind
+		}
+
+		var backendNamespace *gatewayapi_v1.Namespace
+		if backendRef.Namespace != nil && *backendRef.Namespace != "" {
+			backendNamespace = backendRef.Namespace
+		} else {
+			backendNamespace = ptr.To(gatewayapi_v1.Namespace(routeNamespace))
+		}
+
+		policyTargetRef := gatewayapi_v1alpha2.PolicyTargetReferenceWithSectionName{
+			PolicyTargetReference: gatewayapi_v1alpha2.PolicyTargetReference{
+				Group:     backendRefGroup,
+				Kind:      backendRefKind,
+				Name:      backendRef.Name,
+				Namespace: backendNamespace,
+			},
+			SectionName: ptr.To(gatewayapi_v1alpha2.SectionName(service.Weighted.ServicePort.Name)),
+		}
+
+		var upstreamTLS *UpstreamTLS
+		// Check to see if there is any BackendTLSPolicy matching this service and service port
+		backendTLSPolicy, found := p.source.LookupBackendTLSPolicyByTargetRef(policyTargetRef)
+		if found {
+			var caSecrets []*Secret
+			for _, certRef := range backendTLSPolicy.Spec.TLS.CACertRefs {
+				if certRef.Group != "" {
+					continue
```

Comment:
we could add some coverage for this stuff but really it is covered by the `LookupBackendTLSPolicyByTargetRef` helper tbh, so maybe we can leave it or even drop this check?

Reply:
I think it makes sense to drop the check since you are right it is covered by the helper.